csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -77,6 +77,72 @@ void AllocationOrderInferencer::handle(UnaryOp* op) {
   propagateAllocationOrder(in, out);
 }
 
+// BroadcastOp propagation:
```

Comment:
note to @zasdfgbnm 

I kept the old implementation here, i.e. pushing all broadcast iterdomain to outer. I'm imagining this allows us to have better allocation order, i.e. more iter domains staying closer to each other, so better vectorization?!